fdbserver/DataDistributionTracker.actor.cpp

```diff
@@ -707,11 +737,19 @@ Future<Void> shardMerger(DataDistributionTracker* self,
 	    .detail("LastLowBandwidthStartTime", lastLowBandwidthStartTime)
 	    .detail("ShardCount", shardCount);
 
-	if (mergeRange.begin < systemKeys.begin) {
-		self->systemSizeEstimate -= systemBytes;
+	if (!SERVER_KNOBS->EMERGENCY_DISABLE_DATA_MOVE) {
+		if (mergeRange.begin < systemKeys.begin) {
+			self->systemSizeEstimate -= systemBytes;
+		}
+		restartShardTrackers(self, mergeRange, ShardMetrics(endingStats, lastLowBandwidthStartTime, shardCount));
+		self->shardsAffectedByTeamFailure->defineShard(mergeRange);
+	} else {
+		TraceEvent(SevWarnAlways, "SkippedDataMove", self->distributorId)
+		    .detail("DataMovePriority", SERVER_KNOBS->PRIORITY_MERGE_SHARD)
+		    .detail("TargetRange", mergeRange);
+
 	}
-	restartShardTrackers(self, mergeRange, ShardMetrics(endingStats, lastLowBandwidthStartTime, shardCount));
-	self->shardsAffectedByTeamFailure->defineShard(mergeRange);
+
 	self->output.send(RelocateShard(mergeRange, SERVER_KNOBS->PRIORITY_MERGE_SHARD));
```

Comment:
How does this RelocateShard request get rejected by DDQueue?

Reply:
it doesn't, it will go through without changing any internal data-structure, and ignored in moveKeys()

Reply:
I see. Thanks